### DISCUSSION TITLE
Xray Android

### DISCUSSION THREAD

**Discussion by mglwolf** [XTLS/Xray-core#106] [Q&A]
Hi Guy,
I noticed, there's a android version of Xray, May I have more detail about that?
1, can It running on a non-rooted device.
2, is this server version?

**Comment by RPRX** [comment]
1. Google Termux
2. Client & Server

**Comment by mglwolf** [comment]
so you mean, it can be run on termux right?
for a server, can I copy othe config file to android? (I have install Xray on google cloud.)
and any dependency required(could you tell me the requirement)?  Im intrested about this.
Thank you
Br,
Van Wong

**Comment by RPRX** [comment]
1. Correct.
2. Correct.
3. No dependency required. (Golang, after all)

**Reply by mglwolf** [reply]
great! so all protocol can be run on it? I mean vless+tls or vless+xtls or something?

**Comment by mglwolf** [comment]
do I need to install ngix as well?

**Comment by RPRX** [comment]
Just configure it as what's needed as server side.

**Reply by mglwolf** [reply]
Hi rprx,
I use this sample:https://github.com/XTLS/Xray-examples/blob/main/VLESS-TCP-TLS%20(minimal%20by%20rprx)/config_server.json
the enviroment: termux, golang, nginx, android 8(non-rooted)
I changed:
1, the port to 1443, because my device is non-rooted,
2, the uuid
3, added my self-signed crt and key
I ran test command: ./xray -test -config ./config_server.json, it replied "configuration ok". then I ran ./xray -c config ./config_server.jason and checked the program has listened on port 1443. but the client cannot connect this server, pls advise.

**Reply by RPRX** [reply]
AllowInsecure at client side.

**Reply by mglwolf** [reply]
2020/12/28 12:54:02 [Info] transport/internet/tcp: listening TCP on 0.0.0.0:1443
2020/12/28 12:54:02 [Warning] core: Xray 1.1.4 started
2020/12/28 12:54:10 [Info] [2354279466] proxy/vless/inbound: firstLen = 0
2020/12/28 12:54:10 [Info] [2354279466] app/proxyman/inbound: connection ends > proxy/vless/encoding: failed to read request version > EOF
2020/12/28 12:54:10 [Info] [2450730463] proxy/vless/inbound: firstLen = 0
2020/12/28 12:54:10 [Info] [2450730463] app/proxyman/inbound: connection ends > proxy/vless/encoding: failed to read request version > EOF

**Reply by mglwolf** [reply]
Finally. issue fixed,
1, added "domainStrategy": "UseIP"
2, added dns

Thank you rprx.

**Comment by mglwolf** [comment]
well, let me try, Thank you!

**Reply by us254** [reply]
This method can be used in Termux and involves obtaining the kernel 1.8.1 xray in addition to using the configuration file provided as an example (line 4).
```
 pkg update

pkg upgrade

 pkg install git curl

curl -o config_client.json https://raw.githubusercontent.com/XTLS/Xray-examples/main/VLESS-TCP-XTLS-Vision/config_client.json

pkg install wget

 cd ~

wget https://github.com/XTLS/Xray-core/releases/download/v1.8.1/Xray-android-arm64-v8a.zip

unzip Xray-android-arm64-v8a.zip

chmod +x xray

 ./xray run -c config_client.json

```
```
pkg install nano

nano config_client.json
```

Ctrl+X, then Y, and then Enter
